components-api/src/main/scala/pl/touk/nussknacker/engine/api/typed/typing.scala

```diff
@@ -204,8 +218,10 @@ object typing {
         case javaList: java.util.List[_] =>
           typedClass(obj.getClass, List(unionOfElementTypes(javaList.asScala.toList)))
         case typeFromInstance: TypedFromInstance => typeFromInstance.typingResult
-        case other =>
-          Typed(other.getClass)
+        case string: String => Typed.typedValue(string)
```

Comment:
I think we need to do it a bit differently. StringValue/LongValue were needed, as we don't know what additional data will be. Here the situation is different - we know the type, so we can serialize/deserialize more wisely.